### PR TITLE
feat(dragAndDrop): page.dragAndDrop

### PR DESCRIPTION
**Pull request by JoelEinbinder**
Waiting for the chromium specific changes to land.

Proposed api: `page.dragAndDrop(selector1, selector2, options)`

Also exists on the frame.

Assuming people like the api I will go write some docs.

#7562

### CODE REVIEW

**Comment by mxschmitt** [comment]
@JoelEinbinder whats the status on this?

**Comment by JoelEinbinder** [comment]
Needs a discussion of the api.

File: docs/src/api/class-frame.md [hunk 1]
```diff
@@ -374,6 +374,20 @@ Optional event-specific initialization properties.
...
+
+### param: Frame.dragAndDrop.selector2
+- `selector2` <[string]>
+### option: Frame.dragAndDrop.force = %%-input-force-%%
+
+### option: Frame.dragAndDrop.noWaitAfter = %%-input-no-wait-after-%%
+
+### option: Frame.dragAndDrop.timeout = %%-input-timeout-%%
+
+### option: Frame.dragAndDrop.timeout = %%-input-trial-%%
```
Comment (pavelfeldman) [inline docs/src/api/class-frame.md:389]: trial

File: docs/src/api/class-page.md [hunk 2]
```diff
@@ -811,6 +811,20 @@ Optional event-specific initialization properties.
...
+
+### param: Page.dragAndDrop.selector2
+- `selector2` <[string]>
+### option: Page.dragAndDrop.force = %%-input-force-%%
+
+### option: Page.dragAndDrop.noWaitAfter = %%-input-no-wait-after-%%
+
+### option: Page.dragAndDrop.timeout = %%-input-timeout-%%
+
+### option: Page.dragAndDrop.timeout = %%-input-trial-%%
```
Comment (pavelfeldman) [inline docs/src/api/class-page.md:826]: trial

File: src/server/frames.ts [hunk 3]
```diff
@@ -981,6 +981,24 @@ export class Frame extends SdkObject {
...
+        return handle._retryPointerAction(progress, 'move and down', false, async point => {
+          await this._page.mouse.move(point.x, point.y);
+          await this._page.mouse.down();
+        }, options);
+      }));
+      await dom.assertDone(await this._retryWithProgressIfNotConnected(progress, selector2, async handle => {
+        return handle._retryPointerAction(progress, 'move and up', false, async point => {
+          await this._page.mouse.move(point.x, point.y);
+          await this._page.mouse.up();
+        }, options);
```
Comment (pavelfeldman) [inline src/server/frames.ts:997]: You need to convert timeout to deadline before calling the first part and then convert deadline into timeout and pass it into the remainder here.

File: types/types.d.ts [hunk 4]
```diff
@@ -1425,6 +1425,39 @@ export interface Page {
     timeout?: number;
   }): Promise<void>;
 
+  /**
+   * @param selector1
+   * @param selector2
+   * @param options
+   */
+  dragAndDrop(selector1: string, selector2: string, options?: {
```
Comment (mxschmitt) [inline types/types.d.ts:1433]: Let's call it maybe source and target everywhere? So it also reflects nicer in auto completion. Some editors show the parameter name in the parameter list and selector1 and selector2 might be not idiomatic.

Comment (JoelEinbinder) [reply types/types.d.ts]: done.

**Review by mxschmitt — APPROVED** [review]

**Review by pavelfeldman — APPROVED** [review]

**Pull request merged**